docs/repo/property-pages/how-to-add-a-new-property-page.md

```diff
@@ -0,0 +1,88 @@
+
+## Option 1: XAML file on disk
+
+In this option, the XAML resides in a standalone .xaml file on disk and is included in the end user's project as a specific kind of MSBuild item, `PropertyPageSchema`. Visual Studio reads these items from the project to determine which property pages to show.
+
+This may be an attractive option if you already have a NuGet package or Visual Studio extension that injects MSBuild .props and .targets files into the end user's project, or if you want to use MSBuild `Condition`s to control when the property page is available to a project. 
+
+### Step 1: Define the XAML file
+
+Add a new XAML file named "MyPropertyPage.xaml" to your project. Depending on how the file is created you may end up with a `<Page>` item in your project but this is not what we want as we're not using the file to describe a piece of WPF UI.
+
+Update your project to replace the `<Page>` item with one of the following:
+
+- SDK-style projects:
+  ``` xml
+  <None Update="MyPropertyPage.xaml">
+    <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+  </None>
+  ```
+- Non-SDK-style projects:
+  ``` xml
+  <None Include="MyPropertyPage.xaml">
+    <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+  </None>
+  ```
+
+Now VS won't do anything with this file but copy it to the output directory when you build.
+
+### Step 2 (optional): Add the Microsoft.Build.Framework package
+
+Use the NuGet Package Manager to add the Microsoft.Build.Framework package to your project. This is an optional step, but it will allow the XAML editor to find the [Rule](https://docs.microsoft.com/en-us/dotnet/api/microsoft.build.framework.xamltypes.rule) type (and related types) and provide code completion, tool tips, Go to Definition, and other features while you type.
+
+### Step 3: Describe the property page
+
+Replace the contents of MyPropertyPage.xaml with the following:
+
+```xml
+<?xml version="1.0" encoding="utf-8" ?>
+<Rule Name="MyPropertyPage"
+      Description="A description of my project properties page."
+      DisplayName="My Properties"
+      PageTemplate="generic"
+      Order="500"
+      xmlns="http://schemas.microsoft.com/build/2009/properties">
+
+  <Rule.DataSource>
+      <DataSource Persistence="ProjectFile"
+                  SourceOfDefaultValue="AfterContext"
+                  HasConfigurationCondition="False" />
+  </Rule.DataSource>
+
+  <StringProperty Name="MyProperty"
+                  DisplayName="My property"
+                  Description="A property that writes to the project file." />
+
+</Rule>
+```
+
+The format of the file is described in detail in [Property Specification](property-specification.md), but the most important points are:
+- The `Name` must be unique.
+- The `PageTemplate` attribute must have the value `"generic"`.
+
+You should now be able to build and see the MyPropertyPage.xaml copied as-is to the output directory.
+
+### Step 4: Define the `PropertyPageSchema` item
```

Comment:
Consider moving this step after step 1, or merging into step 1, so that all project file manipulation happens at the same time.